server/src/main/scala/org/http4s/server/logging/ServerLogging.scala

```diff
@@ -0,0 +1,46 @@
+package org.http4s
+package server
+package logging
+
+import java.time.format.DateTimeFormatter
+import java.time.{Clock, OffsetDateTime}
+
+import org.http4s.headers.Authorization
+
+abstract class ServerLogging[F[_]] {
+  def enabled: Boolean
+  def logRequestResponse(request: Request[F], response: Response[F]): Unit
```

Comment:
If we pass a byte count here, we could also log the size of chunked transfer encodings.
If we pass an elapsed time here, we could log the elapsed time.
If we throw every ad hoc thing anyone might want to log into this signature, this signature will become unwieldy.

:thinking:

Reply:
Yeah, I thought about having logRequestResponse take all parameters it needed directly, but this would mean that every instance of ServerLogging would be strictly forced to work with these parameters.